src/routes/message.js

```diff
@@ -0,0 +1,78 @@
+const express = require('express');
+const mongo = require('mongodb');
+
+const Message = require('../schemas/message');
+
+const router = new express.Router;
+
+router.post('/',async (req, res) => {
+
+    const newMessage = new Message({
+        ...req.body
+    });
+    try{
+        await newMessage.save();
+        res.status(200).send(newMessage);
+    }catch(error){
+        console.log(error);
+        res.status(400).send(error);
+    }
+
+})
+
+router.get('/get', async (req,res) => {
+    const id = req.body.id;
+    try {
+        const o_ID = await new mongo.ObjectID(id);
+        const message = await Message.findById(o_ID);
+        if(!message){
+            res.status(400).send('Error: message with corresponding ID does not exist');
+        }else{
+            res.status(200).send(message);  
+        }
+
+    } catch (error) {
+        res.status(400).send('Error: ID length not 12');//this error message may not be correct
+    }
+
+})
+
+router.delete('/delete', async (req,res) => {
+    const id = req.body.id;
+    
+    try{
+        const o_ID = await new mongo.ObjectID(id);
+        var message = await Message.findByIdAndRemove(o_ID,{useFindAndModify:false});
+        res.status(200).send('delete success');
+    }catch(error){
+        console.log(error);
+        res.status(400).send('Error: something went wrong');
+    }
+})
+
+router.patch('/edit', async (req,res) => {
+    const id = req.body.id;
+    const updateText = req.body.text;
+    try {
+        const o_ID = await new mongo.ObjectID(id);
+        var message = await Message.findByIdAndUpdate(o_ID,{text:updateText},{useFindAndModify:false});
+        var message = await Message.findById(o_ID);
```

Comment:
Why did you declare this a second time? If you are reassigning you don't need to redeclare the variable. findByIdAndUpdate returns the updated object.

Reply:
findByIdAndUpdate does not return the updated object, it returns the query
https://mongoosejs.com/docs/api/model.html#model_Model.findByIdAndUpdate

Reply:
I wanted the response to returned the updated object

Reply:
Query is a promise like object. It resolves to the updated object.
```
Finds a matching document, updates it according to the update arg, passing any options, and returns the found document (if any) to the callback. The query executes if callback is passed
```

Reply:
If the object it returned is not updated you can also just use find one and update the object and call save. Then you can return that object you found.